internal/middlewares/access_managment.go

```diff
@@ -0,0 +1,180 @@
+package middlewares
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/golang-jwt/jwt"
+
+	"git.epam.com/epm-lstr/epm-lstr-lc/be/internal/authentication"
+	"git.epam.com/epm-lstr/epm-lstr-lc/be/internal/logger"
+	"git.epam.com/epm-lstr/epm-lstr-lc/be/internal/utils"
+)
+
+type AccessManager struct {
+	endpoints       existingEndpoints
+	acceptableRoles []string
+	fullAccessRoles []string
+	accessMap       map[string]map[string][]Path
+}
+
+type existingEndpoints map[string][]string
+
+// NewAccessManager creates new access manager with admin access to all endpoints
+func NewAccessManager(roles, fullAccessRoles []string, endpoints existingEndpoints) *AccessManager {
+	accessMap := make(map[string]map[string][]Path)
+	return &AccessManager{
+		endpoints:       endpoints,
+		acceptableRoles: roles,
+		fullAccessRoles: fullAccessRoles,
+		accessMap:       accessMap,
+	}
+}
+
+type Path struct {
```

Comment:
Is Path or it's fields used somewhere outside of this package?

Reply:
Done

Reply:
Please pay attention to all exported entities, not to only mentioned lines.